src/NBitcoin/Block.cs

```diff
@@ -66,24 +67,69 @@ public BlockHeader()
             this.SetNull();
         }
 
-        public BlockHeader(string hex)
-            : this(Encoders.Hex.DecodeData(hex))
+        public BlockHeader(byte[] bytes)
         {
+            this.ReadWrite(bytes);
         }
 
-        public BlockHeader(byte[] bytes)
+        public BlockHeader(string hex, Network network)
+            : this(hex, network?.Consensus?.ConsensusFactory ?? throw new ArgumentNullException(nameof(network)))
```

Comment:
What is the purpose of the `?` in `...Consensus?...` and what happens if `network.Consensus` is null?

Could this be changed to:

`this(hex, network?.Consensus ?? throw new ArgumentNullException(nameof(network)))` 

in order to provide an argument exception if Consensus is null?

Reply:
yeah all those methods are not used I deleted them and replaced the code with a BlockHeader.Load
Good spot mate